src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -141,21 +151,113 @@ StageWithCaseData getActiveStage(UUID caseUUID, UUID stageUUID) {
     }
 
     @Transactional
-    Stage createStage(UUID caseUUID, String stageType, UUID teamUUID, UUID userUUID, String emailType, UUID transitionNoteUUID) {
-        log.debug("Creating Stage of type: {}", stageType);
-        Stage stage = new Stage(caseUUID, stageType, teamUUID, userUUID, transitionNoteUUID);
+    public Stage createStage(UUID caseUUID, CreateStageRequest createStageRequest) {
+
+        if (caseUUID == null || createStageRequest == null) {
+            String msg = String.format("Missing required info to create new stage; caseUUID: %s, createStageRequest: %s", caseUUID, createStageRequest);
+            log.error(msg);
+            throw new ApplicationExceptions.EntityCreationException(msg, LogEvent.STAGE_CREATE_FAILURE);
+        }
+
+        log.info("Creating Stage: {} for case {}", createStageRequest, caseUUID);
+
+        // find current active stage to deactivate
+        StageWithCaseData activeStageToBeDeactivated = getCurrentActiveStageWithCaseData(caseUUID);
+
+        UUID currentUserUUID = null;
+        if(activeStageToBeDeactivated != null) {
+            currentUserUUID = activeStageToBeDeactivated.getUserUUID();
+        }
+
+        // Create new stage
+        Stage stage = Stage.fromCreateStageRequest(caseUUID, createStageRequest);
+
         CaseData caseData = caseDataService.getCase(caseUUID);
         calculateDeadlines(stage, caseData);
-        stage.setUserUUID(userUUID);
+
         stageRepository.save(stage);
-        caseDataService.updateCaseData(caseData, stage.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stageType));
-        auditClient.createStage(stage);
         log.info("Created Stage: {}, Type: {}, Case: {}, event: {}", stage.getUuid(), stage.getStageType(), stage.getCaseUUID(), value(EVENT, STAGE_CREATED));
-        notifyClient.sendTeamEmail(caseUUID, stage.getUuid(), teamUUID, caseData.getReference(), emailType);
+
+        StageWithCaseData stageWithCaseData = getStageWithCaseData(caseUUID, stage.getUuid());
+        caseDataService.updateCaseData(caseData, stage.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stage.getStageType()));
+
+        // Update team and user to activate new stage
+        assignTeamAndMemberUserToStage(stageWithCaseData, currentUserUUID);
+
+        // remove old active stage
+        if (activeStageToBeDeactivated != null) {
+            deactivateCurrentActiveStage(caseUUID, activeStageToBeDeactivated);
+        }
+
+        // Update audit and timeline - positioned here maintains timeline consistency.
+        auditClient.createStage(stageWithCaseData);
+
+        // Issue assignment notifications
+        updateAssignmentAudit(stageWithCaseData);
```

Comment:
I'll take a look... I followed the original logic used previously which also made three audit lines in the even that a team and user were specified, however the calls to create the stage, allocate the team and allocate the user were initiated in `hocs-workflow`, I'll see if it can be improved.

Reply:
So checking this one out briefly, if I change it capture all in one go there's knock on consequence in the Audit service and potentially changes to Allocation exports which balloons the scope of this ticket, I suggest we add it to the techdebt ticket list and leave as it for now. Let me know if you're happy with that?